pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -1743,6 +1743,21 @@ func TestValidatePipelineParameterVariables_Failure(t *testing.T) {
 			Paths:   []string{"[0].matrix[b-param].value[0]"},
 		},
 		api: "alpha",
+	}, {
+		name: "invalid task use duplicate parameters",
+		tasks: []PipelineTask{{
+			Name:    "foo-task",
+			TaskRef: &TaskRef{Name: "foo-task"},
+			Params: []Param{{
+				Name: "duplicate-param", Value: ParamValue{Type: ParamTypeString, StringVal: "val1"},
+			}, {
+				Name: "duplicate-param", Value: ParamValue{Type: ParamTypeString, StringVal: "val2"},
+			}},
+		}},
```

Comment:
So let's say I add one more param with the same name `duplicate-param`. It fails with:

```
expected exactly one, got both: [0][1].name, [0][2].name",
```

This is misleading, `ErrMultipleOneOf` might not be the best fit here. We can add a generic error saying:

```
params names must be unique, the same param \""+param.Name+"\" is defined multiple times at: <a list of indexes>
```

Reply:
thanks @pritidesai , I think your suggestion is clearer to the user
i change the error to `params names must be unique, the same param: duplicate-param is defined multiple times at: [0].params[1].name, [0].params[2].name`
do you think this is okay